Sources/ViewControllers/Products/Detail/ProductDetailViewController.swift

```diff
@@ -45,6 +45,16 @@ class ProductDetailViewController: ButtonBarPagerTabStripViewController, DataMan
             }
         }
         setUserAgent()
+
+
+        NotificationCenter.default.addObserver(forName: .productChangesUploaded, object: nil, queue: .main) { [weak self] notif in
+            guard let barcode = notif.userInfo?["barcode"] as? String else {
+                return
+            }
+            if barcode == self?.product.barcode {
+                self?.refreshProduct {}
```

Comment:
@teolemon the current displayed product is reloaded only if the update concerns this product

Reply:
👍